airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/write/bulk_loader/BigQueryObjectStorageFormattingWriter.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.write.bulk_loader
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageFormatConfigurationProvider
+import io.airbyte.cdk.load.data.csv.toCsvRecord
+import io.airbyte.cdk.load.data.dataWithAirbyteMeta
+import io.airbyte.cdk.load.data.withAirbyteMeta
+import io.airbyte.cdk.load.file.csv.toCsvPrinterWithHeader
+import io.airbyte.cdk.load.file.object_storage.ObjectStorageFormattingWriter
+import io.airbyte.cdk.load.file.object_storage.ObjectStorageFormattingWriterFactory
+import io.airbyte.cdk.load.message.DestinationRecordRaw
+import jakarta.inject.Singleton
+import java.io.OutputStream
+
+class BigQueryObjectStorageFormattingWriter(
+    private val stream: DestinationStream,
+    outputStream: OutputStream,
+    private val rootLevelFlattening: Boolean
+) : ObjectStorageFormattingWriter {
+
```

Comment:
I feel like there's an easier implementation, something like this

(not entirely sure if the `by csvFormattingWriter` thing works, but like 80% certain it should)
```kotlin
class BQFW(stream, outputStream) : OSFW by csvFormattingWriter {
  private val csvFormattingWriter = CsvFormattingWriter(stream, outputStream, rootLevelFlattening = true)
  
  override fun accept(record) {
    record.rawData.record.emittedAt *= 1000
    csvFormattingWriter.accept(record)
  }
}
```